front/src/utils/helpers.ts

```diff
@@ -0,0 +1,84 @@
+import { useState, useEffect } from 'react';
+import 'moment/locale/fr';
+
+/**
+ * Fonctions utiles pour les cartes des différents projets
+ */
+
+/**
+ * Debounce input fields
+ */
+export const useDebounce = (value: string | number, delay: number) => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+  return debouncedValue;
+};
+
+/**
+ * Debounce input fields
+ */
+export const useDebouncedFunc = (
+  value: string | number,
+  delay: number,
+  func: (newValue: number) => void
+) => {
```

Comment:
Same here, you can use a special type
```suggestion
export const useDebouncedFunc = <T = (number | null) | (string | null)> (
  value: T,
  delay: number,
  func: (newValue: T) => void
) => {
```

Reply:
need to talk about it